docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,559 @@
+
+Below you can find examples how `ngraph::Function` can be created:
+
+~~~~~~~~~~~~~{.cpp}
+// Basic example with explicit Result operation creation
+// Create opset3::Parameter operation with static shape
+auto data = std::make_shared<ngraph::opset3::Parameter>(ngraph::element::f32, ngraph::Shape{3, 1, 2});
+// Create opset3::Constant operation with value
+auto divide_constant = ngraph::opset3::Constant::create(ngraph::element::f32, ngraph::Shape{1}, {1.5});
+// Create opset3::Power operation that takes two opset3::Constant operations as input
+auto pow = std::make_shared<ngraph::opset3::Power>(divide_constant,
+                                                   ngraph::opset3::Constant::create(ngraph::element::f32, ngraph::Shape{1}, {-1}));                                                   
+auto mul = std::make_shared<ngraph::opset3::Multiply>(data, pow);
+auto res = std::make_shared<ngraph::opset3::Result>(mul);
+
+auto f = std::make_shared<ngraph::Function>(ngraph::ResultVector{res}, ngraph::ParameterVector{data});
+~~~~~~~~~~~~~
+
+~~~~~~~~~~~~~{.cpp}
+// Advanced example with multioutput operation. Results operation will be created automatically.
+auto data = std::make_shared<ngraph::opset3::Parameter>(ngraph::element::f32, ngraph::Shape{1, 3, 64, 64});
+auto axis_const = ngraph::opset3::Constant::create(ngraph::element::i64, ngraph::Shape{}/*scalar shape*/, {1});
+// Create opset3::Split operation that splits input to three slices across 1st dimension
+auto split = std::make_shared<ngraph::opset3::Split>(data, axis_const, 3);
+auto relu = std::make_shared<ngraph::opset3::Relu>(split->output(1)/*specify explicit output*/);
+// Results operations will be created automatically based on provided OutputVector
+auto f = std::make_shared<ngraph::Function>(ngraph::OutputVector{split->output(0), relu, split->output(2)}, ngraph::ParameterVector{data});
+~~~~~~~~~~~~~
+
+## Transformations types
+
+There are two main transformation types:
+
+`1.` ngraph::pass::FunctionalPass is used for transformations that take entire `ngraph::Function` as input and process it.
+
+~~~~~~~~~~~~~{.cpp}
+// my_transformation.hpp
+// Template for FunctionPass transformation class
+class MyFunctionTransformation: public ngraph::pass::FunctionPass {
+public:
+    MyFunctionTransformation() : FunctionPass() {}
+
+    bool run_on_function(std::shared_ptr<ngraph::Function> f) override;
+};
+
+// my_transformation.cpp
+#include "my_transformation.hpp"
+
+bool ngraph::pass::MyFunctionTransformation::run_on_function(std::shared_ptr<ngraph::Function> f) {
+    // Transformation code
+    return false;
+}
+~~~~~~~~~~~~~
+
+Using ngraph::FunctionPass you need to override `run_on_function` method where you will write transformation code. Return value must be `true` if original function has changed during transformation otherwise it must be `false`. For transformation API please follow [Working with ngraph::Function] section.
```

Comment:
`original function has changed` What dose this mean?

Reply:
explanation was added